USING_DOCKER.md

```diff
@@ -0,0 +1,137 @@
+# Using the Dockerfile
+
+The `Dockerfile` in this repo builds a Docker container
+that has current versions of all the development tools.
+Using Docker means you do not have to concern
+yourself that these tools will conflict with
+other software on your computer.
+
+## Development cycle
+
+The developer loop when using Docker is substantially the same as
+if the tools had been installed directly.
+
+All the source files live in the `prql` directory on your machine.
+As you edit the source, the tools (wrapped in the Docker container)
+watch those directories and re-run
+so you can see your results instantly.
+
+When you exit the Docker container (say, at the end of the development
+session), the `prql` directory on the local machine contains the
+latest files.
+You can use `git` to pull or to push the `prql` directory as normal.
+
+To do all this, build the Docker container and start it
+as described in the **Installation** section.
+Then read the separate steps in **Running components under Docker**
+for each component you wish to work on.
+
+## Installation
+
+First install Docker on your computer,
+using one of the many guides on the internet.
+
+Next build the Docker container with the following commands.
+_(It will take some time while Docker pulls in all the
+necessary developer tools.)_
+
+```bash
+cd <top-level-PRQL-directory>
+docker build -t prql .
+```
+
+_Optional:_ To perform a minimal suite of tests
+before each new commit to git, install the `pre-commit` program
+on the machine that hosts your Docker container.
+Then (one time) run `pre-commit install-hooks` to set up
+your `git` client to use the `.pre-commit-config.yaml` config file.
+These tests will run each time you commit so you can avoid silly mistakes.
```

Comment:
Interesting... That's another argument for separating `DEVELOPMENT.md` and `USING_DOCKER.md` - at least for the moment. That section in `DEVELOPMENT.md` tells _what_ the tests do, without telling _how_ install it (because that happens automatically with `cargo test`. 

When you're using Docker, you need to set it up separately (see the next item).